pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -530,15 +540,20 @@ func (vs *VSphere) DiskIsAttached(volPath string, nodeName k8stypes.NodeName) (b
 		}
 		vm, err := vs.getVMByName(ctx, nodeName)
 		if err != nil {
+			if vclib.IsNotFound(err) {
+				glog.Infof("Node %q does not exist, disk %s will be detached automatically.", nodeName, volPath)
+				// make the disk as detached and return false without error.
+				return false, nil
+			}
 			glog.Errorf("Failed to get VM object for node: %q. err: +%v", vSphereInstance, err)
 			return false, err
 		}
-		nodeExist, err := vm.Exists(ctx)
+		isActive, err := vm.IsActive(ctx)
 		if err != nil {
-			glog.Errorf("Failed to check whether node %q exist. err: %+v", vSphereInstance, err)
+			glog.Errorf("Failed to check whether node %q is active. err: %+v", vSphereInstance, err)
 			return false, err
 		}
-		if !nodeExist {
+		if !isActive {
 			glog.Errorf("DiskIsAttached failed to determine whether disk %q is still attached: node %q is powered off",
```

Comment:
Same as commented for DisksAreAttached. When node VM is powered off, we can still check if disks are attached to the Node VM, and return actual state of the disk.